user-guide/io.rst

```diff
@@ -6,5 +6,191 @@ I/O and file systems
+
+Lustre
+^^^^^^
+The ARCHER2 ``/work`` file systems use Lustre as a parallel file system technology. The Lustre file system provides POSIX semantics (changes on one node are immediately visible on other nodes) and can support very high data rates for appropriate I/O patterns.
+
+Striping
+^^^^^^^^
+One of the main factors leading to the high performance of Lustre file systems is the ability to stripe data across multiple Object Storage Targets (OSTs) in a round-robin fashion. Files are striped when the data is split up in chunks that will then be stored on different OSTs across the Lustre system. Striping might improve the I/O performance because it increases the available bandwith since multiple processes can read and write the same files simultaneously. However striping can also increase the overhead. Choosing the right striping configuration is key to obtain high performance results.
+
+Users have control of a number of striping settings on Lustre file systems. Although these parameters can be set on a per-file basis they are usually set on directory where your output files will be written so that all output files inherit the settings.
+
+
+Default configuration
+""""""""""""""""""""""
+ The ``/work`` file systems on ARCHER2 have the same default stripe settings:
+
+* A default stripe count of -1
+* A default stripe size of 1 MiB (1048576 bytes)
+  
+These settings have been chosen to provide a good compromise for the wide variety of I/O patterns that are seen on the system but are unlikely to be optimal for any one particular scenario.
+The Lustre command to query the stripe settings for a directory (or file) is ``lfs getstripe``. For example, to query the stripe settings of an already created directory ``res_dir``:
+
+::
+   
+   [user@archer2]$ lfs getstripe res_dir/
+   res_dir
+   stripe_count:   1 stripe_size:    1048576 stripe_offset:  -1 
+   
+Setting Custom Striping Configurations
+""""""""""""""""""""""""""""""""""""""
+Users can set stripe settings for a directory (or file) using the ``lfs setstripe`` command. The options for ``lfs setstripe`` are:
+
+* ``[--stripe-count|-c]`` to set the stripe count; 0 means use the system default (usually 1) and -1 means stripe over all available OSTs.
+* ``[--stripe-size|-s]`` to set the stripe size; 0 means use the system default (usually 1 MB) otherwise use k, m or g for KB, MB or GB respectively
+* ``[--stripe-index|-i]`` to set the OST  index (starting at 0) on which to start striping for this file.  An index of -1  allows the  MDS  to choose the starting index and it is strongly recommended, as this allows space and load balancing to  be  done  by the  MDS  as  needed. 
+
+For example, to set a stripe size of 4 MiB for the existing directory ``res_dir``, along with maximum striping count you would use:
+
+::
+
+   [user@archer2]$ lfs setstripe -s 4m -c -1 res_dir/
+
```

Comment:
Add a note that new files will inherit stripe settings from the directory but that if you change the striping of a directory then any file in that directory are not re-striped to match the new settings.